src/streamlink/plugins/picarto.py

```diff
@@ -1,128 +1,222 @@
+                'online': bool,
+            }],
+        }),
+    })
+    stream_schema = validate.Schema(validate.any({'error': str}, {
+        'source': [{
+            'type': str,
+            'url': validate.url(),
+        }],
+    }), validate.get('source'))
+    vod_schema = validate.Schema({
+        'data': {
+            'video': validate.any(None, {
+                'id': str,
+                'title': str,
+                'file_name': str,
+                'channel': {
+                    'name': str,
+                },
+            }),
+        },
+    }, validate.get('data'), validate.get('video'))
+
+    author = None
+    category = None
+    title = None
 
     @classmethod
     def can_handle_url(cls, url):
-        return cls._url_re.match(url) is not None
-
-    def _create_hls_stream(self, server, channel, token):
-        streams = HLSStream.parse_variant_playlist(self.session,
-                                                   self.HLS_URL.format(
-                                                       server=server,
-                                                       channel=channel,
-                                                       token=token),
-                                                   verify=False)
-        if len(streams) > 1:
-            log.debug("Multiple HLS streams found")
-            return streams
-        elif len(streams) == 0:
-            log.warning("No HLS streams found when expected")
-            return {}
+        return cls.url_re.match(url) is not None
+
+    def get_author(self):
+        return self.author
+
+    def get_category(self):
+        return self.category
+
+    def get_title(self):
+        return self.title
+
+    def parse_proxy_url(self, purl):
+        '''Adapted from UStreamTV plugin (ustreamtv.py)'''
+
+        proxy_options = {}
+        if purl:
+            p = urlparse(purl)
+            proxy_options['proxy_type'] = p.scheme
+            proxy_options['http_proxy_host'] = p.hostname
+            if p.port:
+                proxy_options['http_proxy_port'] = p.port
+            if p.username:
+                proxy_options['http_proxy_auth'] = \
+                    (unquote_plus(p.username), unquote_plus(p.password or ''))
+        return proxy_options
+
+    def get_stream(self, data, live=True, server=None):
+        # Proxy support adapted from the UStreamTV plugin (ustreamtv.py)
+        proxy_url = self.session.get_option('https-proxy')
+        if proxy_url is None:
+            proxy_url = self.session.get_option('http-proxy')
+        proxy_options = self.parse_proxy_url(proxy_url)
+        if proxy_options.get('http_proxy_host'):
+            log.debug('Using proxy ({0}://{1}:{2})'.format(
+                proxy_options.get('proxy_type') or 'http',
+                proxy_options.get('http_proxy_host'),
+                proxy_options.get('http_proxy_port') or 80))
+
+        if live and server:
+            wss_url = self.wss_url.format(
```

Comment:
```
    wss_url = 'wss://{0}/stream/json_{1}'
    live_wss_path = 'golive%2B{0}.js'
    vod_wss_server = 'recording-eu-1.picarto.tv'
    vod_wss_path = '{0}.js'
```

seems unnecessarily complicated

just use something like `wss_url = f'wss://{server}/stream/json_golive%{data}.js'`

Reply:
Hmm, it appears that with an f-string I'd have to take the live and VOD URLs out of the class space and put them directly into the code.  I don't know if that's a good thing or better to leave them defined at the top with the other URLs?

How about defining one for each of live and VOD:
```
live_wss_url = 'wss://{server}/stream/json_golive%{data}.js'
vod_wss_url = 'wss://{server}/stream/json_{data}.js'
```
and then just use:
```
        if live and server:
            wss_url = self.live_wss_url.format(server=server, data=data)
        else:
            server = self.vod_wss_server
            wss_url = self.vod_wss_url(server=server, data=data)
```
instead?  It seems clean, while keeping the URL defs at the top with the others.